crates/runtime/src/search/rrf.rs

```diff
@@ -0,0 +1,571 @@
+
+    fn make_test_table() -> Result<Arc<dyn TableProvider>> {
+        let schema = Arc::new(Schema::new(vec![
+            Field::new("id", DataType::Int64, false),
+            Field::new("content", DataType::Utf8, false),
+            Field::new(
+                "content_embedding",
+                DataType::FixedSizeList(Arc::new(Field::new("item", DataType::Float32, true)), 64),
+                true,
+            ),
+        ]));
+
+        let mut embedded_columns = HashMap::new();
+        embedded_columns.insert(
+            "content".to_string(),
+            EmbeddingColumnConfig {
+                model_name: "test_model".to_string(),
+                vector_size: 64,
+                in_base_table: true,
+                chunker: None,
+            },
+        );
+
+        let embedding_model = Arc::new(
+            Model2Vec::from_params(
+                "minishlab/potion-base-2M",
+                None,
+                None,
+                None,
+                None,
+                None,
+                None,
+            )
+            .expect("Must make embedding model"),
+        );
+
+        let batch = RecordBatch::try_new(
+            Arc::clone(&schema),
+            vec![
+                Arc::new(Int64Array::from_iter_values((0i64..TEST_DATA.len() as i64))),
+                Arc::new(StringArray::from_iter_values(TEST_DATA.iter())),
+                Arc::new(FixedSizeListArray::from_iter_primitive::<
+                    arrow::datatypes::Float32Type,
+                    _,
+                    _,
+                >(
+                    TEST_DATA.iter().map(|s| {
+                        embedding_model
+                            .embed_sync(EmbeddingInput::String(s.to_string()))
+                            .map(|e| e[0].iter().map(|f| Some(*f)).collect::<Vec<Option<_>>>())
+                            .ok()
+                    }),
+                    64,
+                )),
+            ],
+        )?;
+
+        let mem_table = Arc::new(MemTable::try_new(schema, vec![vec![batch]])?);
+        let mut embedding_model_store: HashMap<String, Arc<dyn Embed>> = HashMap::new();
+        embedding_model_store.insert("test_model".to_string(), embedding_model);
+
+        Ok(Arc::new(EmbeddingTable {
+            base_table: mem_table,
+            embedded_columns,
+            embedding_models: Arc::new(RwLock::new(embedding_model_store)),
+        }))
+    }
+
+    async fn make_test_runtime() -> Result<Runtime> {
+        let rt = RuntimeBuilder::new().build().await;
+        rt.df
+            .ctx
+            .state()
+            .config_mut()
+            .set_extension(Arc::clone(&TEST_REQUEST_CONTEXT));
+
+        let test_table = make_test_table()?;
+        rt.df
+            .ctx
+            .register_table("foo", test_table)
+            .expect("Failed to register foo table");
+
+        register_udfs(&rt);
+        Ok(rt)
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
```

Comment:
This tests rrf with multiple subqueries. It only uses `vector_search` calls since the test setup is easier. We need `multi_thread` so we can block during planning.